### DISCUSSION TITLE
Where `artifactory_user` stores generated password?

### DISCUSSION THREAD

**Discussion by piotrminkina** [jfrog/terraform-provider-artifactory#598] [Q&A] [answered]
Hello, I'm using `artifactory_user` resource. I noticed when `password` field is omitted, a random password is generated, but how to get this password? I read the state from the tfstate file and this password is not there. I did not receive it by email either, because I did not configure an SMTP server on my Artifactory instance.

Below is a part of my tfstate file.

```json
    {
      "mode": "managed",
      "type": "artifactory_user",
      "name": "admin",
      "provider": "provider[\"registry.terraform.io/jfrog/artifactory\"]",
      "instances": [
        {
          "schema_version": 0,
          "attributes": {
            "admin": true,
            "disable_ui_access": false,
            "email": "mail@example.com",
            "groups": null,
            "id": "admin",
            "internal_password_disabled": false,
            "name": "admin",
            "password": null,
            "profile_updatable": true
          },
          "sensitive_attributes": [],
          "private": "some-base64-value-here"
        }
      ]
    },
```

I wonder if the `artifactory_user` resource has behaved correctly. Before running Terraform Apply, I forgot to import the existing `admin` user into the tfstate file. Because of this, Terraform wanted to create a new `admin` user, well that failed miserably. In my opinion, this process should have ended with an error, because there was an attempt to create a user that already exists in Artifactory, but there was no mention of this fact in the tfstate file.

Regards
Piotr Minkina

**Comment by alexhung** [comment, marked as answer]
Ok I can confirm this is a bug. Please create a issue so we can work on a fix.

**Reply by alexhung** [reply]
I created issue https://github.com/jfrog/terraform-provider-artifactory/issues/602

**Comment by piotrminkina** [comment]
@alexhung Thanks for created issue. What about the case where an existing 'admin' user has been overwritten by Terraform? In my opinion, Terraform Apply should end with an error about the fact that the user that Terraform wants to create already exists.

**Reply by alexhung** [reply]
The fix should resolve that too. Currently *any* error from the API request is not sent back to Terraform so it doesn't know the request failed.

**Reply by danielmkn** [reply]
@piotrminkina, could you please create an issue with the detailed description, we will take a look. Since the password issue won't be fixed, this problem will require separate attention.

**Comment by danielmkn** [comment]
@piotrminkina, please check #623
